File: Typescript/src/__tests__/squareDigits.test.ts
import { squareDigits, squareDigitsV2 } from '../squareDigits';

describe('Testing squareDigits functions', () => {
    test('squareDigits function', () => {
        expect(squareDigits(5292)).toBe(254814);
        expect(squareDigits(3212)).toBe(9414);
        expect(squareDigits(2112)).toBe(4114);
        expect(squareDigits(0)).toBe(0);
    });

    test('squareDigitsV2 function', () => {
        expect(squareDigitsV2(5292)).toBe(254814);
        expect(squareDigitsV2(3212)).toBe(9414);
        expect(squareDigitsV2(2112)).toBe(4114);
        expect(squareDigitsV2(0)).toBe(0);
    });
});

File: Go/internal/square_digits_test.go
package square_digits

import (
	"testing"
)

type TestCase struct {
	input    int
	expected int
}

var tests = []TestCase{
	{123, 149},
	{25213, 425419},
	{562399, 2536498181},
	{12319571, 14918125491},
}

func TestSquareDigits(t *testing.T) {
	for _, test := range tests {
		result := SquareDigits(test.input)
		if result != test.expected {
			t.Errorf("SquareDigits(%d) = %d; want %d", test.input, result, test.expected)
		}
	}
}

func TestSquareDigitsV2(t *testing.T) {
	for _, test := range tests {
		result := SquareDigitsV2(test.input)
		if result != test.expected {
			t.Errorf("SquareDigitsV2(%d) = %d; want %d", test.input, result, test.expected)
		}
	}
}
